packages/rest/src/rest.server.ts

```diff
@@ -209,6 +226,17 @@ export class RestServer extends Context implements Server, HttpServerLike {
     );
   }
 
+  /**
+   * Set up an express router for all static assets so that middleware for
+   * all directories are invoked at the same phase
```

Comment:
Please note that each new Router layer adds a non-negligible delay to the observed latency (time needed to handle the request) and also reduces the amount of requests that the server can handle every second.

I don't understand what's the purpose of creating a new `express.Router()` dedicated to serving static assets? Is this needed to support `app.static` API? If it is, then I am proposing to use a different implementation where we collect all information provided by `app.static` (e.g. list of arguments passed to every invocation) and then invoke static middleware factory only when constructing the ultimate request handler.

Reply:
I think this should be mostly an implementation detail that we can easily change later. If you prefer, then I am ok to keep the currently proposed implementation for now.

Reply:
It's the need to support `app.static()` api. This is the same express problem that we worked around in LB3 with middleware phases.

Reply:
I understand that and wanted to proposed a different solution to that problem. Let's not waste too much time on this, I can later send a follow-up pull request to update the implementation to what I am envisioning. However, based on my other comment, we may need the extra `express.Router` instance to allow sequence-based implementation, in which case this discussion is not relevant.